src/java/za/co/mip/ablduck/utilities/HTMLGenerator.java

```diff
@@ -0,0 +1,367 @@
+            deps = "<h4>Subclasses</h4>";
+
+            for (String subclass : cls.subclasses) {
+                deps += "<div class='dependency'>" + renderLink(subclass) + this.divEndTag;
+            }
+        }
+        return deps;
+    }
+    
+    private String renderClassComment (SourceJSObject cls) {
+        String classComment = "";
+        
+        if (cls.meta.internal != null) 
+            classComment = renderInternal(cls.meta.internal) + this.brTag;
+                
+        if (cls.meta.deprecated != null)
+            classComment = renderDeprecated("class", cls.meta.deprecated) + this.brTag + classComment;
+               
+        classComment += cls.comment;
+        return classComment;
+    }
+    
+    private String renderMemberDetails (SourceJSObject cls, String memberSection, String memberSectionTitle) {
+        
+        return "<div class='members-section'>"
+                 + "<div class='definedBy'>Defined By</div>"
+                 + "<h3 class='members-title icon-" + memberSection + "'>" + memberSectionTitle + "</h3>"
+                 + "<div class='subsection'>"
+                    + renderMember(cls, memberSection)
+                 + this.divEndTag
+              + this.divEndTag;
+    }
+    
+    private String renderMember (SourceJSObject cls, String memberType) {
+        
+        String memberHTML = "";
+        Boolean first = true;
+        
+        for (MemberObject member : cls.members) {
+            
+            if (member.tagname.equals(memberType)) {
+
+                String firstChild = "";
+                if (first) {
+                    firstChild = "first-child";
+                    first = false;
+                }
+                
+                String inherited;
+                if (member.owner.equals(cls.name))
+                  inherited = "not-inherited";
+                else
+                  inherited = "inherited";
+                
+                String doc = "";
+                if(member.comment != null)
+                    doc = member.comment;
+                
+                String shortdoc = stripHtmlTags(doc);
+                if (shortdoc.length() > 100)
+                    shortdoc = shortdoc.substring(0, 100) + " ..."; 
+                
+                if(shortdoc.length() == 0) {
+                    shortdoc = "&nbsp;";
+                    doc      = shortdoc;
+                }
+                
+                if (member.meta.internal != null) 
+                    doc = renderInternal(member.meta.internal) + this.brTag + doc;
+                
+                if (member.meta.deprecated != null)
+                    doc = renderDeprecated(member.tagname, member.meta.deprecated) + this.brTag + doc;
+               
+                String sig = "";
+                String returnT = "";
+                
+                if ("method".equals(member.tagname)) {
+                    if (member.parameters.size() > 0) {

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1155)